odoo/addons/base/tests/test_res_company.py

```diff
@@ -64,3 +64,14 @@ def test_unlink_company_with_children(self):
         with self.assertRaises(UserError):
             parent_company.unlink()
         self.assertTrue(parent_company.exists())
+
+    def test_create_branch_with_default_parent_id(self):
+        other_company = self.env['res.company'].create({'name': 'Other Company'})
+        parent_id = self.env.company.id
+
+        partner = self.env['res.partner'].browse(parent_id)
+        partner.company_id = other_company
+
+        branch = self.env['res.company'].with_context(default_parent_id=parent_id).create({'name': 'Branch Company'})
+
+        self.assertFalse(branch.partner_id.company_id)
```

Comment:
couldn't we do something like:

```suggestion
        parent_id = self.env.company.id
        branch = self.env['res.company'].with_context(default_parent_id=parent_id).create({'name': 'Branch Company'})
        self.assertFalse(branch.partner_id.parent_id)
```
?